nav2_smoother/src/savitzky_golay_smoother.cpp

```diff
@@ -49,11 +50,16 @@ bool SavitzkyGolaySmoother::smooth(
   double time_remaining = max_time.seconds();
 
   bool success = true, reversing_segment;
+  unsigned int segments_smoothed = 0;
   nav_msgs::msg::Path curr_path_segment;
   curr_path_segment.header = path.header;
 
   std::vector<PathSegment> path_segments = findDirectionalPathSegments(path);
 
+  if (path_segments.empty()) {
```

Comment:
I can see what you're thinking here, but this function always returns at least 1 segment for the 1 segment that could exist for a non-empty path which would be checked by your validation function in the server itself. This may be removed